test/CMakeLists.txt

```diff
@@ -178,8 +178,8 @@ macro(onedpl_add_test test_source_file switch_off_checked_iterators)
 
     set(coal_tests "reduce.pass" "transform_reduce.pass" "count.pass" "sycl_iterator_reduce.pass" "minmax_element.pass")
     if (_test_name IN_LIST coal_tests)
-        onedpl_construct_exec(${test_source_file} ${_test_name} ${switch_off_checked_iterators} "-D_ONEDPL_DETECT_SPIRV_COMPILATION=1")
-        onedpl_construct_exec(${test_source_file} ${_test_name}.coal ${switch_off_checked_iterators} "-D_ONEDPL_DETECT_SPIRV_COMPILATION=0")
+        onedpl_construct_exec(${test_source_file} ${_test_name} ${switch_off_checked_iterators} _ONEDPL_DETECT_SPIRV_COMPILATION=1)
```

Comment:
What is the rationale for removing `-D` for these defines here? According to the CMake documentation for [target_compile_definitions](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html), the `-D` should automatically be removed.

> Any leading -D on an item will be removed.

Reply:
Does the `"` wrapping get in the way here or is that removed by cmake in the passing of and evaluating the argument?

Its possible that the only cmake problem was the lack of `${...}`

Reply:
I took a second look and the quotes here are not affecting correctness, so I will revert this change (along with removing -D).

Reply:
I have reverted these unnecessary changes. The only CMake change that remains is the bugfix to properly use the `custom_define` variable.